common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -394,6 +392,11 @@ private void record0(Object hint) {
             }
         }
 
+        boolean dispose() {
+            clear();
+            return allLeaks.remove(this, LeakEntry.INSTANCE);
```

Comment:
I noticed this when I did my CL, but hadn't come up with a solution.   Apparently allLeaks was never cleared, even since 2013!.

Reply:
it was as it was done as part of `close()` the problem was that it also cleared the linked-list of records that we tried to access then on `toString()` and so was not able to print anything 👎